mod_reforged/config/indirect_numeral_adjectives.nut

```diff
@@ -0,0 +1,202 @@
+	ExactDialogSize = false,
+	PartyRange = false,
+	DialogRange = true
+
+	// Data
+	IndefiniteNumerals = [],
+
+	function register( _name, _startsAt, _suffix = "" )
+	{
+		::Reforged.Numerals.IndefiniteNumerals.push({
+			Name = _name,
+			StartsAt = _startsAt,
+			Suffix = _suffix,
+			Range = ""
+		});
+	}
+
+	function setExactPartySize( _bool )
+	{
+		this.ExactPartySize = _bool;
+		this.updateAllParties();
+	}
+
+	function setPartyRange( _bool )
+	{
+		this.PartyRange = _bool;
+		this.updateAllParties();
+	}
+
+
+	// This always needs to be called after all new numerals of your mod are registered.
+	// This will sort all Numerals and generate their ranges automatically
+	function calculateRanges()
+	{
+		this.IndefiniteNumerals.sort(@(a, b) a.StartsAt <=> b.StartsAt);
+
+		local minAmount = 1;
+		local maxAmount = 1;
+
+		for (local i = 1; i < this.IndefiniteNumerals.len(); i++)
+		{
+			maxAmount = (this.IndefiniteNumerals[i].StartsAt - 1);
+			if (minAmount != maxAmount || minAmount != 1)	// We hard-code that the (1) for One would never displayed
+			{
+				this.IndefiniteNumerals[i - 1].Range = "" + minAmount;
+				if (minAmount != maxAmount) this.IndefiniteNumerals[i - 1].Range += "-" + maxAmount;
+			}
+			minAmount = maxAmount + 1;
+		}
+		this.IndefiniteNumerals.top().Range = minAmount + "+";
+	}
+
+	function getNumeralString( _amount, _isWorldParty )
+	{
+		if (this.IndefiniteNumerals.len() == 0) return _amount;
+		if (this.ExactPartySize && _isWorldParty) return _amount;
+		if (this.ExactDialogSize && !_isWorldParty) return _amount;
+
+		local numeralEntry = this.getNumeralEntryByAmount(_amount);
+
+		local numeralString = numeralEntry.Name;
+		if (!_isWorldParty) numeralString += numeralEntry.Suffix;
+		if (this.PartyRange && _isWorldParty || this.DialogRange && !_isWorldParty)
+		{
+			if (numeralEntry.Range != "") numeralString += " (" + numeralEntry.Range + ")";
+		}
+		return numeralString;
+	}
+
+	// Can be used by other mods to adjust individuell entries
+	function getNumeralEntryByName( _name )
+	{
+		foreach (numeral in this.IndefiniteNumerals)
+		{
+			if (numeral.Name == _namet) return numeral;
+		}
+		return null;
+	}
+
+	// Private
+	function updateAllParties()
+	{
+		// I don't know how else to get "every entity on the world map"
+		foreach (worldParty in ::World.getAllEntitiesAtPos(::World.State.m.Player.getPos(), 150000.0))
+		{
+			worldParty.updateStrength();
+		}
+	}
+
+	// Returns an indirect numeral, given an _amount. This requires 'this.IndefiniteNumerals' to be sorted
+	function getNumeralEntryByAmount( _amount )
+	{
+		local arrayPosition = -1;   // Somewhere the integer needs to be translated into an array position (decrementing by 1) so why not here
+		foreach (numeral in this.IndefiniteNumerals)
+		{
+			if (_amount < numeral.StartsAt) break;		// This should never happen for the first entry
+			++arrayPosition;
+		}
+		arrayPosition = ::Math.min(arrayPosition, this.IndefiniteNumerals.len() - 1);
```

Comment:
Because `arrayPosition` is initialized as `-1`, I don't think it can ever be greater than  `this.IndefiniteNumerals.len() - 1`, so this line seems redundant.